geopandas/plotting.py

```diff
@@ -764,20 +744,51 @@ def plot_dataframe(
                     )
                 )
             else:
-                categories = list(legend_kwds.pop("labels"))
+                labels = list(legend_kwds.pop("labels"))
         else:
             fmt = "{:.2f}"
             if legend_kwds is not None and "fmt" in legend_kwds:
                 fmt = legend_kwds.pop("fmt")
 
-            categories = binning.get_legend_classes(fmt)
+            labels = binning.get_legend_classes(fmt)
             if legend_kwds is not None:
                 show_interval = legend_kwds.pop("interval", False)
             else:
                 show_interval = False
             if not show_interval:
-                categories = [c[1:-1] for c in categories]
-        values = np.array(binning.yb)
+                labels = [c[1:-1] for c in labels]
+
+        lowest = values[~nan_idx].min()
+        if lowest > binning.bins[0]:
+            lowest = np.NINF
+        values = pd.cut(
```

Comment:
The reason to use pd.cut instead of the `binning.yb` is to end up with a pandas Categorical? (to take the categorical path below) 
But doesn't do that the "binning" step twice? Since mapclassify already did it as well?

Reply:
Good point. I've changed it to create categorical from codes provided by `binning.yb` directly.